src/hayhooks/server/utils/yaml_utils.py

```diff
@@ -0,0 +1,153 @@
+from typing import Any, TypedDict, Union
+
+import yaml
+from pydantic import BaseModel
+
+from hayhooks.server.exceptions import InvalidYamlIOError
+
+
+class BaseInputOutputResolution(BaseModel):
+    path: str
+    component: str
+    name: str
+    type: Any
+
+
+class InputResolution(BaseInputOutputResolution):
+    required: bool
+
+
+class OutputResolution(BaseInputOutputResolution):
+    pass
```

Comment:
just out of curiosity: is there a specific reason to used `BaseModel`s here instead of dataclasses?